Documentation/Signoffs/Sensors Subsystem Detail Design.md

```diff
@@ -0,0 +1,195 @@
+2.	The robot's light detector shall be able to detect when the start LED turns on.
+3.	The robot's magnetic sensors shall be able to detect the magnetic fields of the Geodinium.
+4.	All of the robot's sensors shall be able to work effectively despite background interference in the competition environment.
+
+Constraints:
+1.	The robot shall be able to count the number of materials collected and deposited.
+2.	The robot shall know its current position.
+3.	The distance sensor shall detect a maximum of 8 ft, or 2.4 m.
+4.	The magnetic sensor shall provide a signal that is larger than the resolution of the Arduino's Analog-to-Digital Converter.
+5.	The light detector shall provide a signal visible through the background noise.
+7.	The robot's general sensor subsystem shall have a user manual that explains functionality and design intent.
+
+## Overview of Proposed Solution
+To meet the listed specifications and constraints, a suite of sensors is composed to solve each of the problems individually.
+This suite includes a Light Detection and Ranging device (LiDAR), photoresistors, Hall Effect sensors, and an Inertial Measurement Unit (IMU).
+The LiDAR provides direct feedback for position.
+It will provide the distance between the robot and the walls of the arena.
+This meets specification 1 and helps with constraint 2.
+The Hall Effect sensors will be used to detect the magnetic field from the magnetic materials.
+This meets specification 3.
+The IMU will be used to get position and will be the main solution for constraint 2.
+The photoresistors will be used to detect the presence of light.
+One photoresistor will be used to detect when the start LED has turned on.
+This meets specification 2.
+The other photoresistors will be used with their own LEDs to count the number of materials.
+This meets constraint 1.
+For specification 4, the sensors most prone to noise from the environment will be the photoresistors.
+They will need to be properly calibrated to make sure that they do not send a signal when they are not supposed to.
+
+## Interface with Other Subsystems
+The IMU and LiDAR will provide positional feedback to the navigation control.
+The photoresistor for the start LED will turn on the robot so that it can start completing its goals.
+This will require activating the controllers, so that they can start running their scripts.
+Each of the controllers will then properly set up and begin running the other subsystems.
+The photoresistors for counting materials and the Hall Effect sensors will send information to the lower-level controller, which is an Arduino Mega.
+This set of sensors will inform the controller about where the materials should be sent.
+
+## Buildable Schematic
+![Sensors Subsystem Schematic](https://github.com/ACruz-42/F24_Team1_CapstoneDemo/blob/aa634817add249cdd91bff9f9b24f9f2ba25e57d/Documentation/Images/Sensors%20Subsystem/Sensors%20Subsystem%20Schematic.svg)
+
```

Comment:
Are you planning to build a PCB to connect all of these sensors? If yes you should add it here.

Reply:
I do not plan on building a PCB. I will use a solderful (or solderable) breadboard.